pandas/core/computation/align.py

```diff
@@ -200,7 +205,9 @@ def reconstruct_object(typ, obj, axes, dtype):
     res_t = np.result_type(obj.dtype, dtype)
 
     if not isinstance(typ, partial) and issubclass(typ, PandasObject):
-        return typ(obj, dtype=res_t, **axes)
+        if name is None:
+            return typ(obj, dtype=res_t, **axes)
+        return typ(obj, dtype=res_t, name=name, **axes)
```

Comment:
Shouldn't `name=None` also be OK? i.e. Can't this always be `return typ(obj, dtype=res_t, name=name, **axes)`?

Reply:
Unfortunately no. In some cases `typ` is `DataFrame` (in particular cases like `pd.eval("df + df")`) in which case passing anything as `name=` (including `None`) would raise a `TypeError`